src/components/checkbox/checkbox.tsx

```diff
@@ -113,7 +119,8 @@ export class Checkbox {
                 disabled={this.disabled || this.readonly}
                 label={this.label}
                 helperText={this.helperText}
-                checked={this.checked}
+                checked={this.checked || this.indeterminate}
```

Comment:
We want the MDC checkbox checked when indeterminate to force the transition indeterminate->unchecked.
If not, the MDC checkbox will both remove the indeterminate state and toggle checked upon interaction.
For an example of this, see the example at https://kradio3.github.io/react-mdc-web/components/checkbox/ and:

1. check the checkbox
2. click "make indeterminate"
3. click the checkbox to remove indeterminate state
4. checkbox gets the unchecked state
5. click "make indeterminate"
6. click the checkbox to remove the indeterminate state
7. checkbox gets the checked state

By forcing `checked={true}` will force the wanted unchecked state when clicking an indeterminate checkbox.

Reply:
After a second thought, maybe we don't want to limit the checkbox like this? 🤔  If I remove the change on this line a consumer can do as they want, just like one may do with the MDC checkbox. Do you agree about removing this change is better @adrianschmidt ? (less opinionated)

Reply:
I've got to look at the effects of either choice, and I'm holding a trainee workshop today so might not get to that until later, but my gut instinct is to go with _more_ opinionated 😄

After all, we want not just looks, but also behaviour, to be streamlined and consistent when using Lime Elements.

Reply:
In that case I think this is ready for review again. Let me know if I need to move some things in my commits. The boyscout commit is there so I'm hoping it's like you asked for.